scripts/cnode-helper-scripts/cncli.sh

```diff
@@ -797,10 +807,203 @@ cncliPTsendslots() {
   done
 }
 
+#################################
+# epochdata table load process  #
+#################################
+
+getCurrNextEpoch() {
+  getNodeMetrics
+  curr_epoch=${epochnum}
+  next_epoch=$((curr_epoch + 1))
+}
+
+runCurrentEpoch() {
+  getKoiosData
+  getCurrNextEpoch
+  echo "Processing current epoch: ${curr_epoch}"
+  stake_param_curr="--active-stake ${active_stake_set} --pool-stake ${pool_stake_set}"
+
+  ${CNCLI} leaderlog ${cncliParams} --consensus "${consensus}" --epoch="${curr_epoch}" --ledger-set current ${stake_param_curr} |
+  jq -r '[.epoch, .epochNonce, .poolId, .sigma, .d, .epochSlotsIdeal, .maxPerformance, .activeStake, .totalActiveStake] | @csv' |
+  sed 's/"//g' >> "$tmpcsv"
+}
+
+runNextEpoch() {
+  getKoiosData
+  getCurrNextEpoch
+  echo "Processing next epoch: ${next_epoch}"
+  stake_param_next="--active-stake ${active_stake_mark} --pool-stake ${pool_stake_mark}"
+
+  ${CNCLI} leaderlog ${cncliParams} --consensus "${consensus}" --epoch="${next_epoch}" --ledger-set next ${stake_param_next} |
+  jq -r '[.epoch, .epochNonce, .poolId, .sigma, .d, .epochSlotsIdeal, .maxPerformance, .activeStake, .totalActiveStake] | @csv' |
+  sed 's/"//g' >> "$tmpcsv"
+}
+
+runPreviousEpochs() {
+  [[ -z ${KOIOS_API} ]] && return 1
+
+  local one_or_many="${EPOCH}"
+  if [[ -n "$1" && "$1" != "all" ]]; then
+    one_or_many="$1"
+  elif [[ "${subarg}" != "all" && -n "${subarg}" ]]; then
+    one_or_many="${subarg}"
+  fi
```

Comment:
I dont see the purpose of this one. 
In processSingleEpoch() when runPreviousEpochs is called, just pass the epoch to run for as a param to runPreviousEpochs.
Do the same in processAllEpochs(), runPreviousEpochs is called for each epoch in array so can just as well pass the current epoch in loop along to runPreviousEpochs.

Reply:
applied, thank you